Does not build using gcc 6

Hi,

I have not yet tested the 2.3 test release but the Debian packaged version ov 2.2.6 fails to build from source when using gcc version 6 as you can read in the Debian bug report:

    https://bugs.debian.org/811866

> [ 35%] Building CXX object CMakeFiles/HYPHYGTK.dir/src/gui/HYModelWindow.cpp.o
> /usr/bin/c++   -DGDK_PIXBUF_ENABLE_BACKEND -D_HYPHY_LIBDIRECTORY_=\"/usr/lib/hyphy\" -D_SLKP_LFENGINE_REWRITE_ -D_SLKP_USE_SSE_INTRINSICS -D__AFYP_REWRITE_BGM__ -D__HYPHYCURL__ -D__HYPHY_64__ -D__HYPHY_GTK__ -D__UNIX__ -I/<<BUILDDIR>>/hyphy-2.2.6+dfsg/src/core/include -I/<<BUILDDIR>>/hyphy-2.2.6+dfsg/src/lib/Link -I/<<BUILDDIR>>/hyphy-2.2.6+dfsg/src/new/include -I/<<BUILDDIR>>/hyphy-2.2.6+dfsg/src/gui/include -I/<<BUILDDIR>>/hyphy-2.2.6+dfsg/src/gui/include/Components -I/<<BUILDDIR>>/hyphy-2.2.6+dfsg/src/gui/include/WindowClasses -I/<<BUILDDIR>>/hyphy-2.2.6+dfsg/contrib/gtest-1.7.0/include -I/<<BUILDDIR>>/hyphy-2.2.6+dfsg/tests/gtests -I/usr/lib/openmpi/include -I/usr/lib/openmpi/include/openmpi -I/usr/include/gtk-2.0 -I/usr/include/freetype2 -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/include/atk-1.0 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/lib/x86_64-linux-gnu/gtk-2.0/include -I/<<BUILDDIR>>/hyphy-2.2
> .6+dfsg/src/gui/gtk/include -I/<<BUILDDIR>>/hyphy-2.2.6+dfsg/src/gui/gtk/include/Components -I/Developer/Headers/FlatCarbon  -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2    -g -fsigned-char -O3 -msse3   -o CMakeFiles/HYPHYGTK.dir/src/gui/HYModelWindow.cpp.o -c /<<BUILDDIR>>/hyphy-2.2.6+dfsg/src/gui/HYModelWindow.cpp
> /<<BUILDDIR>>/hyphy-2.2.6+dfsg/src/gui/HYModelWindow.cpp: In member function 'virtual bool _HYModelWindow::ProcessEvent(_HYEvent*)':
> /<<BUILDDIR>>/hyphy-2.2.6+dfsg/src/gui/HYModelWindow.cpp:860:40: error: no matching function for call to '_Formula::_Formula(_String&, NULL, bool)'
>              _Formula f (cText,nil,false);
>                                         ^
> 
> In file included from /<<BUILDDIR>>/hyphy-2.2.6+dfsg/src/core/include/parser.h:54:0,
>                  from /<<BUILDDIR>>/hyphy-2.2.6+dfsg/src/core/include/batchlan.h:44,
>                  from /<<BUILDDIR>>/hyphy-2.2.6+dfsg/src/gui/include/Components/HYTableComponent.h:13,
>                  from /<<BUILDDIR>>/hyphy-2.2.6+dfsg/src/gui/include/WindowClasses/HYModelWindow.h:13,
>                  from /<<BUILDDIR>>/hyphy-2.2.6+dfsg/src/gui/HYModelWindow.cpp:41:
> /<<BUILDDIR>>/hyphy-2.2.6+dfsg/src/core/include/formula.h:88:5: note: candidate: _Formula::_Formula(_PMathObj, bool)
>      _Formula (_PMathObj, bool isAVar = false);
>      ^~~~~~~~
> 
> /<<BUILDDIR>>/hyphy-2.2.6+dfsg/src/core/include/formula.h:88:5: note:   candidate expects 2 arguments, 3 provided
> /<<BUILDDIR>>/hyphy-2.2.6+dfsg/src/core/include/formula.h:87:5: note: candidate: _Formula::_Formula(_String&, _VariableContainer*, _String*)
>      _Formula (_String&,_VariableContainer* theParent=nil,_String* errorString = nil);
>      ^~~~~~~~
> 
> /<<BUILDDIR>>/hyphy-2.2.6+dfsg/src/core/include/formula.h:87:5: note:   no known conversion for argument 3 from 'bool' to '_String*'
> /<<BUILDDIR>>/hyphy-2.2.6+dfsg/src/core/include/formula.h:86:5: note: candidate: _Formula::_Formula()
>      _Formula (void);
>      ^~~~~~~~
> 
> /<<BUILDDIR>>/hyphy-2.2.6+dfsg/src/core/include/formula.h:86:5: note:   candidate expects 0 arguments, 3 provided
> /<<BUILDDIR>>/hyphy-2.2.6+dfsg/src/core/include/formula.h:79:9: note: candidate: _Formula::_Formula(const _Formula&)
>  class   _Formula   // a computational formula
>          ^~~~~~~~
> 
> /<<BUILDDIR>>/hyphy-2.2.6+dfsg/src/core/include/formula.h:79:9: note:   candidate expects 1 argument, 3 provided
> CMakeFiles/HYPHYGTK.dir/build.make:785: recipe for target 'CMakeFiles/HYPHYGTK.dir/src/gui/HYModelWindow.cpp.o' failed



There might be more failures later in the build sequence.

Kind regards

        Andreas.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -122,6 +122,11 @@ if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_COMPILER_IS_GNUCXX)
     if(${GCC46})
         set(DEFAULT_WARNING_FLAGS "-Wno-int-to-pointer-cast -Wno-conversion-null")
     endif(${GCC46})
+
+    if(${GCC_VERSION} VERSION_GREATER 6.0 OR ${GCC_VERSION} VERSION_EQUAL 6.0)
+        set(DEFAULT_COMPILE_FLAGS "-fno-strict-aliasing -fpermissive")
+    endif(${GCC_VERSION} VERSION_GREATER 6.0 OR ${GCC_VERSION} VERSION_EQUAL 6.0)
+
     
     PCL_CHECK_FOR_AVX()
     if(${HAVE_AVX_EXTENSIONS})

